### ISSUE TITLE
`npm install` fails without `phantomjs-prebuilt`

### ISSUE THREAD

**Issue by sebastiendecraene** [notable/notable#87]
The npm install failed, but I found the problem and I had to execute `npm install phantomjs-prebuilt` for it to install successfully. Not really an issue but it's for people who are getting the same error :+1:  
We could add the `"phantomjs-prebuilt": "^2.1.16",` to the package.json.

**Comment by fabiospampinato** [comment]
I haven't seen other people reporting this, does any of the current dependencies have `phantomjs-prebuilt` as a peer dependency maybe?

**Comment by sebastiendecraene** [comment]
@fabiospampinato 
Well I couldn't quite figure it out in the stacktrace but I tried it twice in a different folder aswell,
Here is the stack trace: 
```json
seba@seba:~/Desktop/notable$ sudo npm install 
npm WARN deprecated transformation-matrix-js@2.7.6: Package no longer supported. Contact support@npmjs.com for more info.
npm ERR! prepareGitDep 1> 
npm ERR! prepareGitDep > phantomjs-prebuilt@2.1.16 install /home/seba/.npm/_cacache/tmp/git-clone-f96fb8f5/node_modules/phantomjs-prebuilt
npm ERR! prepareGitDep > node install.js
npm ERR! prepareGitDep 
npm ERR! prepareGitDep PhantomJS not found on PATH
npm ERR! prepareGitDep Download already available at /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
npm ERR! prepareGitDep Verified checksum of previously downloaded file
npm ERR! prepareGitDep Extracting tar contents (via spawned process)
npm ERR! prepareGitDep Removing /home/seba/.npm/_cacache/tmp/git-clone-f96fb8f5/node_modules/phantomjs-prebuilt/lib/phantom
npm ERR! prepareGitDep Copying extracted folder /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1546013174226/phantomjs-2.1.1-linux-x86_64 -> /home/seba/.npm/_cacache/tmp/git-clone-f96fb8f5/node_modules/phantomjs-prebuilt/lib/phantom
npm ERR! prepareGitDep 
npm ERR! prepareGitDep 2> npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
npm ERR! prepareGitDep npm WARN deprecated nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
npm ERR! prepareGitDep Phantom installation failed { Error: EACCES: permission denied, link '/tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1546013174226/phantomjs-2.1.1-linux-x86_64' -> '/home/seba/.npm/_cacache/tmp/git-clone-f96fb8f5/node_modules/phantomjs-prebuilt/lib/phantom'
npm ERR! prepareGitDep   errno: -13,
npm ERR! prepareGitDep   code: 'EACCES',
npm ERR! prepareGitDep   syscall: 'link',
npm ERR! prepareGitDep   path: '/tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1546013174226/phantomjs-2.1.1-linux-x86_64',
npm ERR! prepareGitDep   dest: '/home/seba/.npm/_cacache/tmp/git-clone-f96fb8f5/node_modules/phantomjs-prebuilt/lib/phantom' } Error: EACCES: permission denied, link '/tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1546013174226/phantomjs-2.1.1-linux-x86_64' -> '/home/seba/.npm/_cacache/tmp/git-clone-f96fb8f5/node_modules/phantomjs-prebuilt/lib/phantom'
npm ERR! prepareGitDep npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm ERR! prepareGitDep npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! prepareGitDep 
npm ERR! prepareGitDep npm ERR! code ELIFECYCLE
npm ERR! prepareGitDep npm ERR! errno 1
npm ERR! prepareGitDep npm ERR! phantomjs-prebuilt@2.1.16 install: `node install.js`
npm ERR! prepareGitDep npm ERR! Exit status 1
npm ERR! prepareGitDep npm ERR! 
npm ERR! prepareGitDep npm ERR! Failed at the phantomjs-prebuilt@2.1.16 install script.
npm ERR! prepareGitDep npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! prepareGitDep 
npm ERR! prepareGitDep npm ERR! A complete log of this run can be found in:
npm ERR! prepareGitDep npm ERR!     /home/seba/.npm/_logs/2018-12-28T16_06_17_130Z-debug.log
npm ERR! prepareGitDep 
npm ERR! code ENOPACKAGEJSON
npm ERR! package.json Non-registry package missing package.json: codemirror@git://github.com/fabiospampinato/CodeMirror.git.
npm ERR! package.json npm can't find a package.json file in your current directory.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/seba/.npm/_logs/2018-12-28T16_06_17_200Z-debug.log

```

**Comment by tuchandra** [comment]
I've run into the same issue (installing on Debian running on a Samsung Chromebook Plus). In my case, however, `npm install phantomjs-prebuilt` fails because my machine is running arm64, so I had to run `sudo apt-get install phantomjs` first to install it from my OS package manager. I also needed `sudo apt-get install build-essential` to avoid an error about not having `make` installed, and `sudo apt-get install libnss3` to avoid another downstream error that I forgot.

To install NPM itself, I had to follow [instructions from NodeSource](https://github.com/nodesource/distributions/blob/master/README.md#deb), which itself required me to run `sudo apt-get install gnupg` to succeed.

The ultimate goal was just to build and run from source, as described in the comment on #85, but unlike the issue creator, running `./releases/linux-unpacked/notable` works fine for me.

**Comment by fabiospampinato** [comment]
I'm not quite sure why you guys need to install phantomjs. I'd be great if you guys could help pin-point the root cause of the issue so that we can solve this.

**Comment by tuchandra** [comment]
I'm reinstalling from source again and this is happening on the current commit on master (b218179). The error message happens upon running `npm install`. I'm not particularly sure what dependency is requiring PhantomJS, but this is the error message:

```
npm ERR! prepareGitDep 1> 
npm ERR! prepareGitDep > phantomjs-prebuilt@2.1.16 install /home/tchandra104/.npm/_cacache/tmp/git-clone-b5001fe9/node_modules/phantomjs-prebuilt
npm ERR! prepareGitDep > node install.js
npm ERR! prepareGitDep 
npm ERR! prepareGitDep PhantomJS not found on PATH
npm ERR! prepareGitDep 
npm ERR! prepareGitDep 2> npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
npm ERR! prepareGitDep npm WARN deprecated nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
npm ERR! prepareGitDep Unexpected platform or architecture: linux/arm64
npm ERR! prepareGitDep It seems there is no binary available for your platform/architecture
npm ERR! prepareGitDep Try to install PhantomJS globally
npm ERR! prepareGitDep npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm ERR! prepareGitDep npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm64"})
npm ERR! prepareGitDep 
npm ERR! prepareGitDep npm ERR! code ELIFECYCLE
npm ERR! prepareGitDep npm ERR! errno 1
npm ERR! prepareGitDep npm ERR! phantomjs-prebuilt@2.1.16 install: `node install.js`
npm ERR! prepareGitDep npm ERR! Exit status 1
npm ERR! prepareGitDep npm ERR! 
npm ERR! prepareGitDep npm ERR! Failed at the phantomjs-prebuilt@2.1.16 install script.
npm ERR! prepareGitDep npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! prepareGitDep 
npm ERR! prepareGitDep npm ERR! A complete log of this run can be found in:
npm ERR! prepareGitDep npm ERR!     /home/tchandra104/.npm/_logs/2019-01-06T01_59_09_332Z-debug.log
npm ERR! prepareGitDep 
npm ERR! premature close

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tchandra104/.npm/_logs/2019-01-06T01_59_09_713Z-debug.log
```

I can post the entire debug output if you'd like.

**Comment by fabiospampinato** [comment]
Can you guys still reproduce this from the master branch?

**Comment by tuchandra** [comment]
`npm install` worked fine (with several warnings, but whatever), and so did everything until `npm run tutorial`. Then I got the following:

```
tchandra104@penguin:~/notable$ npm run tutorial

> notable@1.3.0 tutorial /home/tchandra104/notable
> cd ./resources/tutorial && zip -FS -r ../../src/renderer/template/dist/tutorial.zip . -x '*.DS_Store'

sh: 1: zip: not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! notable@1.3.0 tutorial: `cd ./resources/tutorial && zip -FS -r ../../src/renderer/template/dist/tutorial.zip . -x '*.DS_Store'`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the notable@1.3.0 tutorial script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tchandra104/.npm/_logs/2019-03-14T00_55_51_456Z-debug.log
```

which looks to me like I'm just missing `zip` on my terminal. But this is just creating the tutorial, so it's not a huge deal.

Running `npm run dev` worked and let me open the app.

**tl;dr things are fine**

Quick note: [CONTRIBUTING.md](https://github.com/notable/notable/blob/master/.github/CONTRIBUTING.md) still has a reference to the repo under your account - I don't believe it made a difference, but it should likely be updated to the new link.

**Comment by fabiospampinato** [comment]
> which looks to me like I'm just missing zip on my terminal.

Yeah that command is not cross-platform.

> tl;dr things are fine

Good, I'm closing this then. I think there might have been a dependency that required phantomjs and maybe that was failing on your systems for some reasons 🤔 

> Quick note: CONTRIBUTING.md still has a reference to the repo under your account - I don't believe it made a difference, but it should likely be updated to the new link.

Good catch! I'm updating that right away, thanks.
